doc/examples/test_serialization.cpp

```diff
@@ -0,0 +1,112 @@
+#include "test.h"
+#include <stdint.h>
+#include <sys/mman.h>
+
+namespace Thread {
+  cothread_t host;
+  cothread_t cpu;
+  cothread_t apu;
+}
+
+namespace Buffer {
+  uint8_t cpu[65536];
+  uint8_t apu[65536];
+}
+
+namespace Memory {
+  uint8_t* buffer;
+}
+
+struct CPU {
+  static auto Enter() -> void;
+  auto main() -> void;
+  auto sub() -> void;
+  auto leaf() -> void;
+} cpu;
+
+struct APU {
+  static auto Enter() -> void;
+  auto main() -> void;
+  auto sub() -> void;
+  auto leaf() -> void;
+} apu;
+
+auto CPU::Enter() -> void {
+  while(true) cpu.main();
+}
+
+auto CPU::main() -> void {
+  printf("2\n");
+  sub();
+}
+
+auto CPU::sub() -> void {
+  co_switch(Thread::apu);
+  printf("4\n");
+  leaf();
+}
+
+auto CPU::leaf() -> void {
+  int x = 42;
+  co_switch(Thread::host);
+  printf("6\n");
+  co_switch(Thread::apu);
+  printf("8 (%d)\n", x);
+  co_switch(Thread::host);
+}
+
+auto APU::Enter() -> void {
+  while(true) apu.main();
+}
+
+auto APU::main() -> void {
+  printf("3\n");
+  sub();
+}
+
+auto APU::sub() -> void {
+  co_switch(Thread::cpu);
+  printf("7\n");
+  leaf();
+}
+
+auto APU::leaf() -> void {
+  co_switch(Thread::cpu);
+}
+
+auto main() -> int {
+  Memory::buffer = (uint8_t*)mmap(
+    (void*)0x10'0000'0000, 2 * 65536,
+    PROT_READ | PROT_WRITE | PROT_EXEC,
+    MAP_FIXED | MAP_PRIVATE | MAP_ANONYMOUS, -1, 0
+  );
+  Memory::buffer[0] = 42;
+  printf("%p (%u)\n", Memory::buffer, Memory::buffer[0]);
+
+  Thread::host = co_active();
+  Thread::cpu = co_derive((void*)(Memory::buffer + 0 * 65536), 65536, CPU::Enter);
```

Comment:
Maybe `co_serializable()` should also be demonstrated here?

Reply:
Sorry, referred to the wrong line, I meant at the `memcpy`

Reply:
That's a good point. I also noticed `co_serializable()` wasn't mentioned in the docs, so I added it there too.